pkg/storage/bucket/s3/config.go

```diff
@@ -47,22 +41,16 @@ type HTTPConfig struct {
 	Transport http.RoundTripper `yaml:"-"`
 }
 
-// RegisterFlagsWithPrefix registers the flags for s3 storage with the provided prefix
-func (cfg *HTTPConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
-	cfg.Config.RegisterFlagsWithPrefix(prefix+"s3.", f)
-}
-
 // Config holds the config options for an S3 backend
 type Config struct {
```

Comment:
@JoaoBraveCoding would you be able to add STSEndpoint to this config?
the thanos store supports this and would add support for minio STS auth flow.

Reply:
Yes, no problem from my side. I'll do it once https://github.com/grafana/loki/pull/11132 merges since it will likely be the first to go. Then I'll need to rebase this one and I'll add your request